src/main/kotlin/it/pagopa/ecommerce/payment/requests/client/NodoPerPmClient.kt

```diff
@@ -28,16 +28,17 @@ public class NodoPerPmClient(
       .body(Mono.just(request), CheckPositionDto::class.java)
       .retrieve()
       .onStatus(HttpStatus::isError) { clientResponse ->
-        clientResponse.bodyToMono(String::class.java).flatMap { errorResponseBody: String ->
-          Mono.error(ResponseStatusException(HttpStatus.BAD_GATEWAY, errorResponseBody))
-        }
+        Mono.error(
```

Comment:
too generic, only when we have a KO in the body response must we return 422....in this way we remap all the errors of the nodo pagoPA to 422.....for example if the node returns 500 the service must return 502... .and so on

Reply:
however, the specs of the `checkPosition` needs to be reviewed

Reply:
For now in case of error 5XX I return 502 in case of 4XX I return 422 for you okay ? @infantesimone

Reply:
@infantesimone this clientResponse.bodyToMono(String::class.java). made the client go in exception